macosx/BlocklistDownloader.mm

```diff
@@ -76,57 +76,52 @@ - (void)cancelDownload
 {
     [fViewController setFinished];
 
-    [fDownload cancel];
+    [fSession invalidateAndCancel];
 
     [BlocklistScheduler.scheduler updateSchedule];
 
     fBLDownloader = nil;
 }
 
-//using the actual filename is the best bet
-- (void)download:(NSURLDownload*)download decideDestinationWithSuggestedFilename:(NSString*)filename
-{
-    [fDownload setDestination:[NSTemporaryDirectory() stringByAppendingPathComponent:filename] allowOverwrite:NO];
-}
-
-- (void)download:(NSURLDownload*)download didCreateDestination:(NSString*)path
-{
-    fDestination = path;
-}
-
-- (void)download:(NSURLDownload*)download didReceiveResponse:(NSURLResponse*)response
+- (void)URLSession:(NSURLSession *)session 
+      downloadTask:(NSURLSessionDownloadTask *)downloadTask 
+      didWriteData:(int64_t)bytesWritten 
+ totalBytesWritten:(int64_t)totalBytesWritten 
+totalBytesExpectedToWrite:(int64_t)totalBytesExpectedToWrite
 {
     fState = BLOCKLIST_DL_DOWNLOADING;
 
-    fCurrentSize = 0;
-    fExpectedSize = response.expectedContentLength;
+    fCurrentSize = totalBytesWritten;
+    fExpectedSize = totalBytesExpectedToWrite;
 
     [fViewController setStatusProgressForCurrentSize:fCurrentSize expectedSize:fExpectedSize];
 }
 
-- (void)download:(NSURLDownload*)download didReceiveDataOfLength:(NSUInteger)length
+- (void)URLSession:(NSURLSession *)session 
+              task:(NSURLSessionTask *)task 
+didCompleteWithError:(NSError *)error
 {
-    fCurrentSize += length;
-    [fViewController setStatusProgressForCurrentSize:fCurrentSize expectedSize:fExpectedSize];
-}
-
-- (void)download:(NSURLDownload*)download didFailWithError:(NSError*)error
-{
-    [fViewController setFailed:error.localizedDescription];
+    if (error)
+    {
+        [fViewController setFailed:error.localizedDescription];
+    }
 
     [NSUserDefaults.standardUserDefaults setObject:[NSDate date] forKey:@"BlocklistNewLastUpdate"];
     [BlocklistScheduler.scheduler updateSchedule];
 
     fBLDownloader = nil;
 }
 
-- (void)downloadDidFinish:(NSURLDownload*)download
+- (void)URLSession:(NSURLSession *)session 
+      downloadTask:(NSURLSessionDownloadTask *)downloadTask 
+didFinishDownloadingToURL:(NSURL *)location
 {
     fState = BLOCKLIST_DL_PROCESSING;
 
     [fViewController setStatusProcessing];
 
-    NSAssert(fDestination != nil, @"the blocklist file destination has not been specified");
+    fDestination = [NSTemporaryDirectory() stringByAppendingPathComponent:location.lastPathComponent];
```

Comment:
As per: https://developer.apple.com/documentation/foundation/nsurlsessiondownloaddelegate/1411575-urlsession?language=objc

>A file URL for the temporary file. Because the file is temporary, you must either open the file for reading or move it to a permanent location in your app’s sandbox container directory before returning from this delegate method.

I've decided to copy that file to another temp location, to replicate previous behavior.

Maybe it can be simplified after some trials and refactorings, but that's unscoped to this PR.